src/state.rs

```diff
@@ -29,69 +30,81 @@ pub struct State {
 impl State {
     async fn create_render_device(
         window: &Window,
-    ) -> (wgpu::Surface, wgpu::Adapter, wgpu::Device, wgpu::Queue) {
-        let instance = wgpu::Instance::new(wgpu::BackendBit::PRIMARY);
+    ) -> (
+        wgpu::SurfaceConfiguration,
+        wgpu::Surface,
+        wgpu::Adapter,
+        wgpu::Device,
+        wgpu::Queue,
+    ) {
+        let instance = wgpu::Instance::new(wgpu::Backends::PRIMARY);
         let render_surface = unsafe { instance.create_surface(window) };
+
         let adapter = instance
             .request_adapter(&wgpu::RequestAdapterOptions {
-                power_preference: wgpu::PowerPreference::HighPerformance,
+                power_preference: wgpu::PowerPreference::default(),
```

Comment:
This is HighPerformance because it'd otherwise default to the internal graphics on laptops, which is generally not what you want. Could be a configuration thing in the future I guess

```suggestion
                power_preference: wgpu::PowerPreference::HighPerformance,
```